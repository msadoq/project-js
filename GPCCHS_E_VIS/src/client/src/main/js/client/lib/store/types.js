/**
 * Namespaces: HSC, HSS, WS
 * Domains: WINDOW, PAGE, VIEW, CD, EDITOR
 * Verbs: ADD, REMOVE, UN/MOUNT, SET, UPDATE, OPEN, CLOSE
 */

export const HSC_UPDATE_STATUS = 'HSC_UPDATE_STATUS';
export const HSC_SET_WINDOWS_AS_OPENED = 'HSC_SET_WINDOWS_AS_OPENED';
export const HSC_SET_WORKSPACE_AS_OPENED = 'HSC_SET_WORKSPACE_AS_OPENED';
export const HSC_UPDATE_LAST_CACHE_INVALIDATION = 'HSC_UPDATE_LAST_CACHE_INVALIDATION';
export const HSS_WS_REMOVE = 'HSS_WS_REMOVE';
export const HSS_WS_UPDATE_STATUS = 'HSS_WS_UPDATE_STATUS';
export const HSS_UPDATE_DOMAINS = 'HSS_UPDATE_DOMAINS';
export const HSS_UPDATE_SESSIONS = 'HSS_UPDATE_SESSIONS';
export const HSS_UPDATE_MASTER_SESSION = 'HSS_UPDATE_MASTER_SESSION';
export const HSC_PLAY = 'HSC_PLAY';
export const HSC_PAUSE = 'HSC_PAUSE';
export const HSC_FOCUS_WINDOW = 'HSC_FOCUS_WINDOW';
export const HSC_BLUR_WINDOW = 'HSC_BLUR_WINDOW';

export const HSC_SEND_PRODUCT_LOG = 'HSC_SEND_PRODUCT_LOG';

export const HSC_OPEN_DIALOG = 'HSC_OPEN_DIALOG';
export const HSC_DIALOG_CLOSED = 'HSC_DIALOG_CLOSED';

export const HSC_OPEN_WIKI_HELPER = 'HSC_OPEN_WIKI_HELPER';

export const WS_ASK_OPEN_VIEW = 'WS_ASK_OPEN_VIEW';
export const WS_ASK_SAVE_VIEW = 'WS_ASK_SAVE_VIEW';
export const WS_ASK_CLOSE_VIEW = 'WS_ASK_CLOSE_VIEW';
export const WS_ASK_RELOAD_VIEW = 'WS_ASK_RELOAD_VIEW';
export const WS_ASK_SAVE_VIEW_AS_MODEL = 'WS_ASK_SAVE_VIEW_AS_MODEL';
export const WS_ASK_OPEN_LINK = 'WS_ASK_OPEN_LINK';
export const WS_VIEW_OPENED = 'WS_VIEW_OPENED';
export const WS_VIEW_TOUCH = 'WS_VIEW_TOUCH';

export const WS_ASK_OPEN_PAGE = 'WS_ASK_OPEN_PAGE';
export const WS_ASK_SAVE_PAGE = 'WS_ASK_SAVE_PAGE';
export const WS_ASK_CLOSE_PAGE = 'WS_ASK_CLOSE_PAGE';
export const WS_PAGE_OPENED = 'WS_PAGE_OPENED';

export const WS_ASK_OPEN_WORKSPACE = 'WS_ASK_OPEN_WORKSPACE';
export const WS_ASK_SAVE_WORKSPACE = 'WS_ASK_SAVE_WORKSPACE';
export const WS_ASK_CLOSE_WORKSPACE = 'WS_ASK_CLOSE_WORKSPACE';
export const WS_WORKSPACE_OPENED = 'WS_WORKSPACE_OPENED';

export const WS_ASK_CLOSE_WINDOW = 'WS_ASK_CLOSE_WINDOW';

export const WS_WORKSPACE_SET_MODIFIED = 'WS_WORKSPACE_SET_MODIFIED';
export const WS_WORKSPACE_UPDATE_DOMAINNAME = 'WS_WORKSPACE_UPDATE_DOMAINNAME';
export const WS_WORKSPACE_UPDATE_SESSIONNAME = 'WS_WORKSPACE_UPDATE_SESSIONNAME';

export const WS_MESSAGE_ADD = 'WS_MESSAGE_ADD';
export const WS_MESSAGE_REMOVING = 'WS_MESSAGE_REMOVING';
export const WS_MESSAGE_REMOVE = 'WS_MESSAGE_REMOVE';
export const WS_MESSAGE_CANCEL_REMOVING = 'WS_MESSAGE_CANCEL_REMOVING';
export const WS_MESSAGE_RESET = 'WS_MESSAGE_RESET';

export const WS_VIEW_SET_OID = 'WS_VIEW_SET_OID';
export const WS_PAGE_SET_OID = 'WS_PAGE_SET_OID';

export const WS_TIMEBAR_CREATE_NEW = 'WS_TIMEBAR_CREATE_NEW';
export const WS_TIMEBAR_ID_UPDATE = 'WS_TIMEBAR_ID_UPDATE';
export const WS_TIMEBAR_UPDATE_CURSORS = 'WS_TIMEBAR_UPDATE_CURSORS';
export const WS_TIMEBAR_UPDATE_VIEWPORT = 'WS_TIMEBAR_UPDATE_VIEWPORT';
export const WS_TIMEBAR_SPEED_UPDATE = 'WS_TIMEBAR_SPEED_UPDATE';
export const WS_TIMEBAR_MODE_UPDATE = 'WS_TIMEBAR_MODE_UPDATE';
export const WS_TIMEBAR_MASTERID_UPDATE = 'WS_TIMEBAR_MASTERID_UPDATE';
export const WS_TIMEBAR_DEFAULTWIDTH_UPDATE = 'WS_TIMEBAR_DEFAULTWIDTH_UPDATE';
export const WS_TIMEBAR_SET_REALTIME = 'WS_TIMEBAR_SET_REALTIME';
export const WS_TIMEBAR_GO_NOW = 'WS_TIMEBAR_GO_NOW';

export const WS_TIMELINE_CREATE_NEW = 'WS_TIMELINE_CREATE_NEW';
export const WS_TIMELINE_REMOVE = 'WS_TIMELINE_REMOVE';
export const WS_TIMELINE_UPDATE_ID = 'WS_TIMELINE_UPDATE_ID';
export const WS_TIMELINE_UPDATE_OFFSET = 'WS_TIMELINE_UPDATE_OFFSET';
export const WS_TIMELINE_UPDATE_COLOR = 'WS_TIMELINE_UPDATE_COLOR';
export const WS_TIMELINE_UPDATE_SESSIONNAME = 'WS_TIMELINE_UPDATE_SESSIONNAME';

export const WS_WINDOW_ADD = 'WS_WINDOW_ADD';
export const WS_WINDOW_CLOSE = 'WS_WINDOW_CLOSE';
export const WS_WINDOW_SET_IS_LOADED = 'WS_WINDOW_SET_IS_LOADED';
export const WS_WINDOW_UPDATE_GEOMETRY = 'WS_WINDOW_UPDATE_GEOMETRY';
export const WS_WINDOW_UPDATE_TITLE = 'WS_WINDOW_UPDATE_TITLE';
export const WS_WINDOW_SET_DISPLAY_HELP = 'WS_WINDOW_SET_DISPLAY_HELP';
export const WS_WINDOW_PAGE_FOCUS = 'WS_WINDOW_PAGE_FOCUS';
export const WS_WINDOW_PAGE_REORDER = 'WS_WINDOW_PAGE_REORDER';
export const WS_WINDOW_MINIMIZE = 'WS_WINDOW_MINIMIZE';
export const WS_WINDOW_RESTORE = 'WS_WINDOW_RESTORE';
export const WS_WINDOW_OPEN_CODE_EDITOR = 'WS_WINDOW_OPEN_CODE_EDITOR';
export const WS_WINDOW_CLOSE_CODE_EDITOR = 'WS_WINDOW_CLOSE_CODE_EDITOR';
export const WS_WINDOW_MOVE_TAB_ORDER = 'WS_WINDOW_MOVE_TAB_ORDER';

export const WS_PAGE_ADD_BLANK = 'WS_PAGE_ADD_BLANK';
export const WS_PAGE_CLOSE = 'WS_PAGE_CLOSE';
export const WS_PAGE_UPDATE_LAYOUT = 'WS_PAGE_UPDATE_LAYOUT';
export const WS_PAGE_TIMEBAR_MOUNT = 'WS_PAGE_TIMEBAR_MOUNT';
export const WS_PAGE_TIMEBAR_UNMOUNT = 'WS_PAGE_TIMEBAR_UNMOUNT';
export const WS_PAGE_UPDATE_TITLE = 'WS_PAGE_UPDATE_TITLE';
export const WS_PAGE_PANELS_LOAD_IN_EDITOR = 'WS_PAGE_PANELS_LOAD_IN_EDITOR';
export const WS_PAGE_PANELS_RESIZE_EDITOR = 'WS_PAGE_PANELS_RESIZE_EDITOR';
export const WS_PAGE_PANELS_MINIMIZE_EDITOR = 'WS_PAGE_PANELS_MINIMIZE_EDITOR';
export const WS_PAGE_PANELS_RESIZE_TIMEBAR = 'WS_PAGE_PANELS_RESIZE_TIMEBAR';
export const WS_PAGE_PANELS_MINIMIZE_TIMEBAR = 'WS_PAGE_PANELS_MINIMIZE_TIMEBAR';
export const WS_PAGE_PANELS_FOCUS_IN_EXPLORER = 'WS_PAGE_PANELS_FOCUS_IN_EXPLORER';
export const WS_PAGE_PANELS_RESIZE_EXPLORER = 'WS_PAGE_PANELS_RESIZE_EXPLORER';
export const WS_PAGE_PANELS_MINIMIZE_EXPLORER = 'WS_PAGE_PANELS_MINIMIZE_EXPLORER';
export const WS_PAGE_UPDATE_DOMAINNAME = 'WS_PAGE_UPDATE_DOMAINNAME';
export const WS_PAGE_UPDATE_SESSIONNAME = 'WS_PAGE_UPDATE_SESSIONNAME';

export const WS_VIEW_ADD_BLANK = 'WS_VIEW_ADD_BLANK';
export const WS_VIEW_CLOSE = 'WS_VIEW_CLOSE';
export const WS_VIEW_REMOVE = 'WS_VIEW_REMOVE';
export const WS_VIEW_RELOAD = 'WS_VIEW_RELOAD';
export const WS_VIEW_MOVE_TO_PAGE = 'WS_VIEW_MOVE_TO_PAGE';
export const WS_VIEW_UPDATE_ENTRYPOINT = 'WS_VIEW_UPDATE_ENTRYPOINT';
export const WS_VIEW_UPDATE_AXIS = 'WS_VIEW_UPDATE_AXIS';
export const WS_VIEW_UPDATE_GRID = 'WS_VIEW_UPDATE_GRID';
export const WS_VIEW_UPDATE_LINK = 'WS_VIEW_UPDATE_LINK';
export const WS_VIEW_TOGGLE_LEGEND = 'WS_VIEW_TOGGLE_LEGEND';
export const WS_VIEW_UPDATE_MARKER = 'WS_VIEW_UPDATE_MARKER';
export const WS_VIEW_UPDATE_PROCEDURE = 'WS_VIEW_UPDATE_PROCEDURE';
export const WS_VIEW_UPDATE_RATIO = 'WS_VIEW_UPDATE_RATIO';
export const WS_VIEW_UPDATE_TITLE = 'WS_VIEW_UPDATE_TITLE';
export const WS_VIEW_UPDATE_TITLESTYLE = 'WS_VIEW_UPDATE_TITLESTYLE';
export const WS_VIEW_UPDATE_BGCOLOR = 'WS_VIEW_UPDATE_BGCOLOR';
export const WS_VIEW_UPDATE_LEGEND = 'WS_VIEW_UPDATE_LEGEND';
export const WS_VIEW_UPDATE_CONTENT = 'WS_VIEW_UPDATE_CONTENT';
export const WS_VIEW_UPDATE_SHOWYAXES = 'WS_VIEW_UPDATE_SHOWYAXES';
export const WS_VIEW_UPDATE_DIMENSIONS = 'WS_VIEW_UPDATE_DIMENSIONS';
export const WS_VIEW_ADD_AXIS = 'WS_VIEW_ADD_AXIS';
export const WS_VIEW_REMOVE_AXIS = 'WS_VIEW_REMOVE_AXIS';
export const WS_VIEW_ADD_ENTRYPOINT = 'WS_VIEW_ADD_ENTRYPOINT';
export const WS_VIEW_REMOVE_ENTRYPOINT = 'WS_VIEW_REMOVE_ENTRYPOINT';
export const WS_VIEW_ADD_GRID = 'WS_VIEW_ADD_GRID';
export const WS_VIEW_REMOVE_GRID = 'WS_VIEW_REMOVE_GRID';
export const WS_VIEW_ADD_LINK = 'WS_VIEW_ADD_LINK';
export const WS_VIEW_REMOVE_LINK = 'WS_VIEW_REMOVE_LINK';
export const WS_VIEW_UPDATE_SHOWLINK = 'WS_VIEW_UPDATE_SHOWLINK';
export const WS_VIEW_ADD_MARKER = 'WS_VIEW_ADD_MARKER';
export const WS_VIEW_REMOVE_MARKER = 'WS_VIEW_REMOVE_MARKER';
export const WS_VIEW_ADD_PROCEDURE = 'WS_VIEW_ADD_PROCEDURE';
export const WS_VIEW_REMOVE_PROCEDURE = 'WS_VIEW_REMOVE_PROCEDURE';
export const WS_VIEW_UPDATE_EDITOR_SEARCH = 'WS_VIEW_UPDATE_EDITOR_SEARCH';
export const WS_VIEW_UPDATE_DOMAINNAME = 'WS_VIEW_UPDATE_DOMAINNAME';
export const WS_VIEW_UPDATE_SESSIONNAME = 'WS_VIEW_UPDATE_SESSIONNAME';
export const WS_VIEWDATA_CLEAN = 'WS_VIEWDATA_CLEAN';
export const WS_VIEW_UPDATE_SORTING = 'WS_VIEW_UPDATE_SORTING';
export const WS_VIEW_HIDE_COL = 'WS_VIEW_HIDE_COL';
export const WS_VIEW_SHOW_COL = 'WS_VIEW_SHOW_COL';
export const WS_VIEW_ADD_COL = 'WS_VIEW_ADD_COL';
export const WS_VIEW_REMOVE_COL = 'WS_VIEW_REMOVE_COL';

export const WS_VIEW_ALARM_COLLAPSE = 'WS_VIEW_ALARM_COLLAPSE';
export const WS_VIEW_ALARM_UNCOLLAPSE = 'WS_VIEW_ALARM_UNCOLLAPSE';
export const WS_VIEW_UPDATE_ALARMDOMAIN = 'WS_VIEW_UPDATE_ALARMDOMAIN';
export const WS_VIEW_UPDATE_ALARMTIMELINE = 'WS_VIEW_UPDATE_ALARMTIMELINE';
export const WS_VIEW_UPDATE_ALARMMODE = 'WS_VIEW_UPDATE_ALARMMODE';
export const WS_VIEW_GMA_ALARM_ACK = 'WS_VIEW_GMA_ALARM_ACK';
export const WS_VIEW_GMA_ALARM_ACK_SUCCESS = 'WS_VIEW_GMA_ALARM_ACK_SUCCESS';
export const WS_VIEW_GMA_ALARM_ACK_FAILURE = 'WS_VIEW_GMA_ALARM_ACK_FAILURE';

export const WS_VIEW_OBA_ALARM_ACK = 'WS_VIEW_OBA_ALARM_ACK';
export const WS_VIEW_OBA_ALARM_ACK_SUCCESS = 'WS_VIEW_OBA_ALARM_ACK_SUCCESS';
export const WS_VIEW_OBA_ALARM_ACK_FAILURE = 'WS_VIEW_OBA_ALARM_ACK_FAILURE';

export const DATA_INCOMING_ARCHIVE = 'DATA_INCOMING_ARCHIVE';
export const DATA_INCOMING_PUBSUB = 'DATA_INCOMING_PUBSUB';
export const DATA_REMOVE_ALL_VIEWDATA = 'DATA_REMOVE_ALL_VIEWDATA';

export const HSC_UPDATE_PATH = 'HSC_UPDATE_PATH';
export const WS_VIEW_UPDATEPATH = 'WS_VIEW_UPDATEPATH';
export const WS_VIEW_UPDATE_ABSOLUTEPATH = 'WS_VIEW_UPDATE_ABSOLUTEPATH';
export const WS_PAGE_UPDATEPATH = 'WS_PAGE_UPDATEPATH';
export const WS_PAGE_UPDATE_ABSOLUTEPATH = 'WS_PAGE_UPDATE_ABSOLUTEPATH';

export const HSC_CLOSE_WORKSPACE = 'HSC_CLOSE_WORKSPACE';
export const HSC_ISWORKSPACE_OPENING = 'HSC_ISWORKSPACE_OPENING';
export const WS_PAGE_SETMODIFIED = 'WS_PAGE_SETMODIFIED';
export const WS_WINDOW_SETMODIFIED = 'WS_WINDOW_SETMODIFIED';
export const WS_VIEW_SETMODIFIED = 'WS_VIEW_SETMODIFIED';
export const WS_VIEW_SETCOLLAPSED = 'WS_VIEW_SETCOLLAPSED';
export const WS_VIEW_SETMAXIMISED = 'WS_VIEW_SETMAXIMISED';

export const WS_MODAL_OPEN = 'WS_MODAL_OPEN';
export const WS_MODAL_CLOSE = 'WS_MODAL_CLOSE';
export const WS_MODAL_CLOSED = 'WS_MODAL_CLOSED';

export const WS_EDITOR_UI_PANEL = 'WS_EDITOR_UI_PANEL';
export const WS_EDITOR_UI_SUBPANEL = 'WS_EDITOR_UI_SUBPANEL';
export const WS_EDITOR_UI_TAB = 'WS_EDITOR_UI_TAB';

export const WS_KNOWNINTERVAL_ADD = 'WS_KNOWNINTERVAL_ADD';
export const WS_KNOWNINTERVAL_DELETE = 'WS_KNOWNINTERVAL_DELETE';
export const RESET_KNOWN_RANGES = 'RESET_KNOWN_RANGES';

export const HSS_UPDATE_DC_STATUS = 'HSS_UPDATE_DC_STATUS';
export const HSS_UPDATE_HEALTH_STATUS = 'HSS_UPDATE_HEALTH_STATUS';
export const HSS_UPDATE_MAIN_STATUS = 'HSS_UPDATE_MAIN_STATUS';
export const HSS_UPDATE_WINDOW_STATUS = 'HSS_UPDATE_WINDOW_STATUS';
export const HSS_UPDATE_LAST_PUBSUB_TIMESTAMP = 'HSS_UPDATE_LAST_PUBSUB_TIMESTAMP';

export const HSC_ASK_OPEN_INSPECTOR = 'HSC_ASK_OPEN_INSPECTOR';
export const HSC_IS_INSPECTOR_DISPLAYING_A_TM = 'HSC_IS_INSPECTOR_DISPLAYING_A_TM';
export const HSC_SET_INSPECTOR_GENERAL_DATA = 'HSC_SET_INSPECTOR_GENERAL_DATA';
export const HSC_DELETE_INSPECTOR_GENERAL_DATA = 'HSC_DELETE_INSPECTOR_GENERAL_DATA';
export const HSC_SET_INSPECTOR_STATIC_DATA = 'HSC_SET_INSPECTOR_STATIC_DATA';
export const HSC_IS_INSPECTOR_STATIC_DATA_LOADING = 'HSC_IS_INSPECTOR_STATIC_DATA_LOADING';
export const HSC_TOGGLE_ALL_INSPECTOR_STATIC_DATA_NODES = 'HSC_TOGGLE_ALL_INSPECTOR_STATIC_DATA_NODES';
export const HSC_UPDATE_INSPECTOR_STATIC_DATA_NODE = 'HSC_UPDATE_INSPECTOR_STATIC_DATA_NODE';
export const HSC_IS_INSPECTOR_STATIC_DATA_NODE_LOADING = 'HSC_IS_INSPECTOR_STATIC_DATA_NODE_LOADING';
export const HSC_IS_INSPECTOR_STATIC_DATA_NODE_TOGGLED = 'HSC_IS_INSPECTOR_STATIC_DATA_NODE_TOGGLED';

export const HSC_UPDATE_FORECAST = 'HSC_UPDATE_FORECAST';

export const HSC_SET_RTE_SESSIONS = 'HSC_SET_RTE_SESSIONS';
export const HSC_SET_RTE_DOMAINS = 'HSC_SET_RTE_DOMAINS';
export const HSC_SET_RTE_CATALOGS = 'HSC_SET_RTE_CATALOGS';
export const HSC_SET_RTE_ITEM_NAMES = 'HSC_SET_RTE_ITEM_NAMES';
export const HSC_OPEN_RTE_ITEM = 'HSC_OPEN_RTE_ITEM';
export const HSC_CLOSE_RTE_ITEM = 'HSC_CLOSE_RTE_ITEM';
export const HSC_SET_RTE_FOCUSED_ITEM = 'HSC_SET_RTE_FOCUSED_ITEM';
export const HSC_IS_RTE_CATALOG_NODE_TOGGLED = 'HSC_IS_RTE_CATALOG_NODE_TOGGLED';
export const HSC_IS_RTE_ITEM_NAME_NODE_TOGGLED = 'HSC_IS_RTE_ITEM_NAME_NODE_TOGGLED';
export const HSC_IS_RTE_ITEM_NODE_TOGGLED = 'HSC_IS_RTE_ITEM_NODE_TOGGLED';
export const HSC_IS_RTE_CATALOG_NODE_ACTIVE = 'HSC_IS_RTE_CATALOG_NODE_ACTIVE';
export const HSC_IS_RTE_ITEM_NAME_NODE_ACTIVE = 'HSC_IS_RTE_ITEM_NAME_NODE_ACTIVE';

// PROFILING
export const HSC_UPDATE_STRESS = 'HSC_UPDATE_STRESS';

// DATA FLOW

export const INCOMING_LAST_DATA = 'INCOMING_LAST_DATA';
export const INCOMING_RANGE_DATA = 'INCOMING_RANGE_DATA';
export const INCOMING_PUBSUB_DATA = 'INCOMING_PUBSUB_DATA';
export const INCOMING_PUBSUBALARM_DATA = 'INCOMING_PUBSUBALARM_DATA';

export const NEW_DATA = 'NEW_DATA';

// Is it a good idea to seprate both ?
export const INJECT_DATA_RANGE = 'INJECT_DATA_RANGE';
export const INJECT_DATA_LAST = 'INJECT_DATA_LAST';

export const VIEWS_NEED_LAST = 'VIEWS_NEED_LAST';
export const VIEWS_NEED_RANGE = 'VIEWS_NEED_RANGE';

export const WS_RETRIEVEDATA_RANGE = 'WS_RETRIEVEDATA_RANGE';
export const WS_RETRIEVEDATA_LAST = 'WS_RETRIEVEDATA_LAST';

// BATCH ACTION
export const BATCH = 'BATCHING_REDUCER.BATCH';
