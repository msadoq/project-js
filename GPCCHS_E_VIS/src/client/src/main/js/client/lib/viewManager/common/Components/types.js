import PropTypes from 'prop-types';

const { func, shape, number, string, bool, arrayOf, oneOfType } = PropTypes;

export const connectedDataType = shape({
  axisId: string,
  digit: number,
  domain: string,
  filter: arrayOf(shape({
    field: string,
    operand: string,
    operator: string,
  })),
  format: string,
  formula: string,
  fieldX: string,
  timeline: string,
  unit: string,
});

export const stateColorType = shape({
  color: string.isRequired,
  condition: shape({
    field: string.isRequired,
    operand: string.isRequired,
    operator: string.isRequired,
  }),
});

export const entryPointType = shape({
  id: string,
  name: string,
  parametric: bool,
  connectedData: connectedDataType.isRequired,
  connectedDataParametric: shape({
    xAxisId: string,
    YAxisId: string,
  }),
  objectStyle: shape(),
  stateColors: arrayOf(stateColorType).isRequired,
  obsolete: bool,
  nonsignificant: bool,
});

export const domainType = shape({
  domainId: number.isRequired,
  name: string.isRequired,
  itemNamespace: string,
  oid: string,
  parentDomainId: number,
});

export const domainsType = arrayOf(domainType.isRequired);

export const timelineType = shape({
  color: string,
  id: string.isRequired,
  kind: string,
  offset: number,
  sessionName: string.isRequired,
  uuid: string.isRequired,
});

export const timelinesType = arrayOf(timelineType.isRequired);

export const sessionType = shape({
  id: number.isRequired,
  name: string.isRequired,
  delta: PropTypes.number,
  missionEpoch: PropTypes.number,
  timestamp: PropTypes.shape({
    ms: PropTypes.number,
    ps: PropTypes.number,
  }),
});

export const sessionsType = arrayOf(sessionType.isRequired);

export const catalogItemType = shape({
  name: string.isRequired,
});

export const catalogType = shape({
  name: string.isRequired,
  items: oneOfType([
    string, // when requesting
    arrayOf(catalogItemType),
  ]),
});

export const reduxFormFieldsType = {
  form: string.isRequired,
};

export const comObjectType = shape({
  name: string.isRequired,
  type: string.isRequired,
});

export const TableConfigurationColumnType = shape({
  title: string.isRequired,
  value: string.isRequired,
  position: number.isRequired,
  displayed: bool.isRequired,
  group: number,
});

export const fieldArrayPropsType = shape({
  push: func,
  remove: func,
  insert: func,
  getAll: func,
});

export const applicationProcessType = shape({
  APId: number,
  name: string,
});

export const applicationProcessesType = arrayOf(applicationProcessType.isRequired);
