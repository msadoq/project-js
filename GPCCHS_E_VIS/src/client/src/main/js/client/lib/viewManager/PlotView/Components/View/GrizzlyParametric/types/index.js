import PropTypes from 'prop-types';

export const axisType = PropTypes.shape({
  id: PropTypes.string.isRequired,
  orient: PropTypes.string.isRequired,
  extents: PropTypes.array.isRequired,
  autoLimits: PropTypes.bool.isRequired,
  showAxis: PropTypes.bool.isRequired,
  showLabels: PropTypes.bool,
  showTicks: PropTypes.bool,
  autoTick: PropTypes.bool,
  tickStep: PropTypes.number,
  showGrid: PropTypes.bool,
  gridStyle: PropTypes.string,
  gridSize: PropTypes.number,
  unit: PropTypes.string,
  label: PropTypes.string.isRequired,
  format: PropTypes.string,
  labelStyle: PropTypes.shape,
  formatAsDate: PropTypes.bool,
});

export const lineType = PropTypes.shape({
  data: PropTypes.objectOf(PropTypes.shape),
  indexes: PropTypes.arrayOf(PropTypes.number),
  id: PropTypes.string.isRequired,
  yAxisId: PropTypes.string.isRequired,
  xAxisId: PropTypes.string.isRequired,
  fill: PropTypes.string,
  lineStyle: PropTypes.string,
  lineSize: PropTypes.number,
  pointSize: PropTypes.number,
  pointStyle: PropTypes.string,
  dataAccessor: PropTypes.string,
  stopInstruction: PropTypes.func,
  xAccessor: PropTypes.func,
  yAccessor: PropTypes.func,
  xTooltipAccessor: PropTypes.func,
  yTooltipAccessor: PropTypes.func,
  colorAccessor: PropTypes.string,
  tooltipFormatter: PropTypes.func,
});

export const labelStyleType = PropTypes.shape({
  bgColor: PropTypes.string,
  color: PropTypes.string,
  align: PropTypes.string,
  font: PropTypes.string,
  italic: PropTypes.bool,
  bold: PropTypes.bool,
  underline: PropTypes.bool,
  size: PropTypes.number,
});

export const divStyleType = PropTypes.shape({
  top: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  left: PropTypes.number,
  right: PropTypes.number,
});
