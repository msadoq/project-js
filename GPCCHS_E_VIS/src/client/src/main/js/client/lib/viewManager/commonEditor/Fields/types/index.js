import { PropTypes } from 'react';

const { shape, number, string, arrayOf } = PropTypes;

export const domainType = shape({
  domainId: number,
  itemNamespace: string,
  name: string,
  oid: string,
  parentDomainId: number,
});

export const domainsType = arrayOf(domainType.isRequired);

export const timelineType = shape({
  color: string,
  id: string,
  kind: string,
  offset: number,
  sessionId: number,
  timelineUuid: string,
});

export const timelinesType = arrayOf(timelineType.isRequired);
