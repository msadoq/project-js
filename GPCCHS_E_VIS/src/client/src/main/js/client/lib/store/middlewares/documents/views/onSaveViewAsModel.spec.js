// ====================================================================
// HISTORY
// VERSION : 1.1.2 : FA : #7145 : 24/07/2017 : Add "make" prefix to each documents middleware
// VERSION : 1.1.2 : FA : #7145 : 24/07/2017 : Write documents/onSaveViewAsModel middleware unit tests
// VERSION : 1.1.2 : DM : #6700 : 03/08/2017 : Merge branch 'dev' into dbrugne-data
// VERSION : 1.1.2 : FA : ISIS-FT-2138 : 05/09/2017 : Page / Views / Workspace are saved with extensions.
// END-HISTORY
// ====================================================================

import _ from 'lodash/fp';
import configureMockStore from 'redux-mock-store';
import makeOnSaveViewAsModel from './onSaveViewAsModel';
import * as types from '../../../types';

const documentManager = {
  saveViewAsModel: (viewId, absolutePath) => ({
    type: 'SAVE_VIEW_AS_MODEL',
    payload: { viewId, absolutePath },
  }),
};

const askSaveViewAsModel = viewId => ({
  type: types.WS_ASK_SAVE_VIEW_AS_MODEL,
  payload: { viewId },
});

const mockStore = configureMockStore([makeOnSaveViewAsModel(documentManager)]);

describe('store:serverProcess:middlewares:documents:makeOnSaveViewAsModel', () => {
  const store = mockStore({
    windows: { w1: { focusPage: 'p1', pages: ['p1'] } },
    pages: { p1: { uuid: 'p1', views: ['v1'] } },
    views: { v1: { uuid: 'v1', type: 'PlotView' } },
  });

  test('save view as model', () => {
    store.dispatch(askSaveViewAsModel('v1'));
    const actions = store.getActions();
    const dialogId = _.last(actions).payload.dialogId;
    store.dispatch({ type: 'HSC_DIALOG_CLOSED', payload: { dialogId, choice: '/the/chosen/one' } });
    expect(store.getActions()).toMatchSnapshot();
  });

  test('save view as model (cancel filepicker)', () => {
    store.dispatch(askSaveViewAsModel('v1'));
    const actions = store.getActions();
    const dialogId = _.last(actions).payload.dialogId;
    store.dispatch({ type: 'HSC_DIALOG_CLOSED', payload: { dialogId } });
    expect(store.getActions()).toMatchSnapshot();
  });
});
