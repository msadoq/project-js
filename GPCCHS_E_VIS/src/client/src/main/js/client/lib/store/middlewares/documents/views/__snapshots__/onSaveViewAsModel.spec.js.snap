// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store:serverProcess:middlewares:documents:makeOnSaveViewAsModel save view as model (cancel filepicker) 1`] = `
Array [
  Object {
    "payload": Object {
      "viewId": "v1",
    },
    "type": "WS_ASK_SAVE_VIEW_AS_MODEL",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "options": Object {
        "defaultPath": "/data/work/gitRepositories/LPISIS/GPCCHS/GPCCHS_E_VIS/src/client/src/main/js/client/lib/serverProcess/documentManager/fixtures",
        "filters": Array [
          Object {
            "extensions": Array [
              "vipv",
            ],
            "name": "PlotView",
          },
        ],
      },
      "type": "save",
      "windowId": "w1",
    },
    "type": "HSC_OPEN_DIALOG",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "choice": "/the/chosen/one",
    },
    "type": "HSC_DIALOG_CLOSED",
  },
  Object {
    "payload": Object {
      "absolutePath": "/the/chosen/one",
      "viewId": "v1",
    },
    "type": "SAVE_VIEW_AS_MODEL",
  },
  Object {
    "payload": Object {
      "viewId": "v1",
    },
    "type": "WS_ASK_SAVE_VIEW_AS_MODEL",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "options": Object {
        "defaultPath": "/data/work/gitRepositories/LPISIS/GPCCHS/GPCCHS_E_VIS/src/client/src/main/js/client/lib/serverProcess/documentManager/fixtures",
        "filters": Array [
          Object {
            "extensions": Array [
              "vipv",
            ],
            "name": "PlotView",
          },
        ],
      },
      "type": "save",
      "windowId": "w1",
    },
    "type": "HSC_OPEN_DIALOG",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {},
    "type": "HSC_DIALOG_CLOSED",
  },
]
`;

exports[`store:serverProcess:middlewares:documents:makeOnSaveViewAsModel save view as model 1`] = `
Array [
  Object {
    "payload": Object {
      "viewId": "v1",
    },
    "type": "WS_ASK_SAVE_VIEW_AS_MODEL",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "options": Object {
        "defaultPath": "/data/work/gitRepositories/LPISIS/GPCCHS/GPCCHS_E_VIS/src/client/src/main/js/client/lib/serverProcess/documentManager/fixtures",
        "filters": Array [
          Object {
            "extensions": Array [
              "vipv",
            ],
            "name": "PlotView",
          },
        ],
      },
      "type": "save",
      "windowId": "w1",
    },
    "type": "HSC_OPEN_DIALOG",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "choice": "/the/chosen/one",
    },
    "type": "HSC_DIALOG_CLOSED",
  },
  Object {
    "payload": Object {
      "absolutePath": "/the/chosen/one",
      "viewId": "v1",
    },
    "type": "SAVE_VIEW_AS_MODEL",
  },
]
`;
